pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -935,22 +949,60 @@ func validateMatchCondition(v *admissionregistration.MatchCondition, opts valida
 	return allErrors
 }
 
-func validateValidation(v *admissionregistration.Validation, paramKind *admissionregistration.ParamKind, opts validationOptions, fldPath *field.Path) field.ErrorList {
+func validateVariable(compiler plugincel.Compiler, v *admissionregistration.Variable, paramKind *admissionregistration.ParamKind, opts validationOptions, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	trimmedName := strings.TrimSpace(v.Name)
+	trimmedExpression := strings.TrimSpace(v.Expression)
+	if len(trimmedName) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("name"), "name is not specified"))
+	}
+	if len(trimmedExpression) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("expression"), "expression is not specified"))
+	}
+	if len(trimmedName) > 0 && len(trimmedExpression) > 0 {
+		if !isCELIdentifier(v.Name) { // enforce original name to avoid trimming later in the controller
```

Comment:
this looks required either way and could return early on failure at the top of `validateVariable`

Reply:
Yes! The identifier check should always be done as long as the name is not empty. Fixed.